packages/protocol/contracts/layer1/verifiers/SP1Verifier.sol

```diff
@@ -78,13 +79,54 @@ contract SP1Verifier is EssentialContract, IVerifier {
 
     /// @inheritdoc IVerifier
     function verifyBatchProof(
-        ContextV2[] calldata, /*_ctxs*/
-        TaikoData.TierProof calldata /*_proof*/
+        ContextV2[] calldata _ctxs,
+        TaikoData.TierProof calldata _proof
     )
         external
-        pure
-        notImplemented
-    { }
+        view
+    {
+        // Extract the necessary data
+        bytes32 aggregation_program = bytes32(_proof.data[0:32]);
+        bytes32 block_proving_program = bytes32(_proof.data[32:64]);
+        bytes memory proof = _proof.data[64:];
```

Comment:
if you use `_proof.data[64:]` directly on line122, maybe you can avoid mem-copy?